apis/python/src/tiledbsc/soma_collection.py

```diff
@@ -1,11 +1,13 @@
 from typing import Optional, List
 
-import tiledb
-
 from .soma_options import SOMAOptions
 from .soma import SOMA
+from .soma_slice import SOMASlice
 from .tiledb_group import TileDBGroup
 
+import tiledb
+import pandas as pd
+
 
 class SOMACollection(TileDBGroup):
```

Comment:
__init__ is failing to pass ctx to the super.__init__

Reply:
example problem:  I have a SOCO with hundreds of large SOMAs in it.  I expected the following to work, but it does not:

```
soco = tiledbsc.SOMACollection("./atlas", ctx=tiledb.Ctx({"py.init_buffer_bytes": 4 * 1024 ** 3}))
sum(len(s.obs.ids()) for s in soco)
```

Because the buffer size is not passed down correctly, this iterator fails and there is no fix:
```
---------------------------------------------------------------------------
TileDBError                               Traceback (most recent call last)
<ipython-input-63-533fc342913f> in <module>
----> 1 sum(len(s.obs.ids()) for s in soco)

<ipython-input-63-533fc342913f> in <genexpr>(.0)
----> 1 sum(len(s.obs.ids()) for s in soco)

~/TileDB-SingleCell/apis/python/src/tiledbsc/annotation_dataframe.py in ids(self)
     74             # TileDB string dims are ASCII not UTF-8. Decode them so they readback
     75             # not like `b"AKR1C3"` but rather like `"AKR1C3"`.
---> 76             retval = A.query(attrs=[], dims=[self.dim_name])[:][self.dim_name].tolist()
     77             return [e.decode() for e in retval]
     78 

tiledb/libtiledb.pyx in tiledb.libtiledb.Query.__getitem__()

tiledb/libtiledb.pyx in tiledb.libtiledb.SparseArrayImpl.subarray()

tiledb/libtiledb.pyx in tiledb.libtiledb.SparseArrayImpl._read_sparse_subarray()

TileDBError: Exceeded maximum retries ('py.max_incomplete_retries': '100') (tiledb/core.cc:1221)

```